@here/harp-mapview/lib/geometry/LodMesh.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MathUtils } from "@here/harp-geoutils";
+import * as THREE from "three";
+
+/**
+ * A mesh that can store multiple versions of a geometry with differents level of detail.
+ * The rendered level of detail can be adapted during runtime.
+ * @internal
+ * @hidden
+ */
+export class LodMesh extends THREE.Mesh {
```

Comment:
Not sure if this is a good idea, b/c if the user modifies the mesh geometry, the changes could suddenly be overridden when the zoomlevel changes.
Can you do some measurements what's the actual difference in number of vertices/triangles for the more detailed mesh. If the difference is not to big, we could just always use the higher LOD

Reply:
Do we have use-cases where the geometry created by harp.gl is changed later?
For mixed LOD to work without cracks in the globe it is important that tiles with different zoom level fit together at the edges, e.g., the tessellation of one level 3 tile needs to be the same as 2x2 level 4 tiles. This needs to be changed depending on display zoom level of the data source.